src/routes/api/users.js

```diff
@@ -1,87 +1,85 @@
-const mongoose = require("mongoose");
-const router = require("express").Router();
-const passport = require("passport");
-const User = mongoose.model("User");
+import express from 'express';
+import passport from 'passport';
 
-router.get("/user", function(req, res, next) {
-    User.findById(req.payload.id)
-        .then(function(user) {
-            if (!user) {
-                return res.sendStatus(401);
-            }
-            return res.json({ user: user.toAuthJSON() });
-        })
-        .catch(next);
+import User from '../../models/User';
+
+const router = express.Router();
+
+router.get('/user', (req, res, next) => {
+  User.findById(req.payload.id)
+    .then((user) => {
+      if (!user) {
+        return res.sendStatus(401);
+      }
+      return res.json({ user: user.toAuthJSON() });
+    })
+    .catch(next);
 });
 
-router.put("/user", function(req, res, next) {
-    User.findById(req.payload.id)
-        .then(function(user) {
-            if (!user) {
-                return res.sendStatus(401);
-            }
+router.put('/user', (req, res, next) => {
+  User.findById(req.payload.id)
+    .then((user) => {
+      if (!user) {
+        return res.sendStatus(401);
+      }
 
-            // only update fields that were actually passed...
-            if (typeof req.body.user.username !== "undefined") {
-                user.username = req.body.user.username;
-            }
-            if (typeof req.body.user.email !== "undefined") {
-                user.email = req.body.user.email;
-            }
-            if (typeof req.body.user.bio !== "undefined") {
-                user.bio = req.body.user.bio;
-            }
-            if (typeof req.body.user.image !== "undefined") {
-                user.image = req.body.user.image;
-            }
-            if (typeof req.body.user.password !== "undefined") {
-                user.setPassword(req.body.user.password);
-            }
+      if (typeof req.body.user.username !== 'undefined') {
```

Comment:
`what is the difference between doing this ```req.body.user.username !== 'undefined'``` and doing ```!req.body.username```. If they serve the same purpose. I think you should use the latter.

Reply:
This has been resolved